### ISSUE TITLE
[Bug]: AC0032 - false positive (the return)

### ISSUE THREAD

**Issue by pibcht** [ALCops/Analyzers#343]
### Analyzer

ApplicationCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

AC0032

### Description

I have the exact same issue as described in #243 but I’m using v0.9.0

<img width="183" height="37" alt="Image" src="https://github.com/user-attachments/assets/dc79b892-fbdd-4a81-a5ce-ac56443a7e1f" />

The original code:
<img width="871" height="364" alt="Image" src="https://github.com/user-attachments/assets/5fe4195f-d948-446f-bb4e-95e030146292" />

After commenting it out:
<img width="641" height="316" alt="Image" src="https://github.com/user-attachments/assets/e97475a9-7f86-43bd-a065-b64da940fee7" />

Maybe I missed something. I read the previous issue, but I couldn’t find a clear answer.

### AL code to reproduce

```al
table 50000 Setup
{
    Access = Internal;
    AllowInCustomizations = Never;
    Caption = 'Setup';
    DataClassification = CustomerContent;
    DataPerCompany = true;
    Extensible = true;
    Permissions = tabledata Setup2 = rim; // <-- HERE

    fields
    {
        field(1; PK; Boolean)
        {
            Access = Internal;
            Caption = 'Primary Key', Locked = true;
            Editable = false;
            InitValue = true;
            NotBlank = true;
        }
        field(10; Hash; Guid)
        {
            Access = Internal;
            Caption = 'Hash', Locked = true;
            Editable = false;
            NotBlank = true;
        }
        field(11; Enabled; Boolean)
        {
            Caption = 'Enabled';
            DataClassification = SystemMetadata;
            ToolTip = 'Enable or disable the functionality.';
        }
    }
    keys
    {
        key(PK; PK) { Clustered = true; }
    }

    internal procedure GetOrInitialize()
    begin
        if this.Get(true) then exit;

        this.Init();
        this.Insert(true);
    end;

    internal procedure Disable()
    begin
        this.Validate(Enabled, false);
        this.Modify(true);
    end;
}

```

### Additional context

Same issue as described here : https://github.com/ALCops/Analyzers/issues/243

**Comment by Arthurvdv** [comment]
This should now be resolved with the release of `v1.0.1` of ALCops.

When you have a chance, could you update and verify whether the issue is fixed?

If everything is working as expected, feel free to close this issue. If you still run into the problem, please leave the issue open and share any additional context, code samples, or reproduction steps that might help me investigate further.

**Comment by pibcht** [comment]
The bug is indeed fixed, thank you very much!
